py/kubeflow/testing/cleanup_ci.py

```diff
@@ -787,6 +788,23 @@ def execute_rpc(rpc):
   """Execute a Google RPC request with retries."""
   return rpc.execute()
 
+# Wait for 'ops' to finish in 'max_wait_mins' or return the remaining ops.
+# operation_resource must implement 'get()' method.
+def wait_ops_max_mins(operation_resource, args, ops, max_wait_mins=15):
```

Comment:
What was the bug in the existing implementation?

Reply:
The bug is that the existing wait logic is inside the for-loop of deployments. Hence there is always one op in `delete_ops`, i.e. the function wait for at least 30secs for each deletion.

I think the intention is to kick off all deletion ops and then wait for them to complete. So in the code below i move the `delete_ops` and wait logic out of the deployment for-loop.